src/hooks/useDetectOutsideClick.tsx

```diff
@@ -0,0 +1,29 @@
+import React, { useState, useEffect } from 'react'
+export const useDetectOutsideClick = (
```

Comment:
this is minor semantics, but I think a more descriptive name for this hook would be `useCloseWhenClickedOutside` (since this does more than just detect an outside click, it is also responsible for managing the opened/closed state)